src/realm/sync/instruction_replication.hpp

```diff
@@ -198,8 +198,6 @@ class TempShortCircuitReplication {
     bool m_was_short_circuited;
 };
 
-constexpr bool SYNC_SUPPORTS_NESTED_COLLECTIONS = false;
```

Comment:
I guess we should not have this set to true before the server is ready to support nested collections. I appreciate that this should be enabled in order to see if the tests pass. Will the test pass when set to false?

Reply:
the tests crash when set to false (we use the legacy server to sync with for the tests)